boards/arm/nucleo_f429zi/nucleo_f429zi.dts

```diff
@@ -96,3 +97,51 @@ arduino_serial: &usart6 {};
 &iwdg {
 	status = "ok";
 };
+
+&flash0 {
+	/*
+	 * For more information, see:
+	 * http://docs.zephyrproject.org/devices/dts/flash_partitions.html
+	 */
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* 128K for bootloader */
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x00020000>;
+			read-only;
+		};
+
+		slot0_partition: partition@20000 {
+			label = "image-0";
+			reg = <0x00020000 0x00040000>;
+		};
+
+		/* backup slot: 256KB */
+		slot1_partition: partition@60000 {
+			label = "image-1";
+			reg = <0x00060000 0x00040000>;
+		};
+
+		/* swap slot: 256KB */
+		scratch_partition: partition@a0000 {
+			label = "image-scratch";
+			reg = <0x000a0000 0x00040000>;
+		};
+
+		/* storage: 128KB */
+		storage_partition: partition@e0000 {
+			label = "storage";
+			reg = <0x000e0000 0x00020000>;
+		};
+
+		/* data partition: 1MB */
```

Comment:
Can you elaborate on this choice as well?
Looking to the big picture, we have a 2MB flash size available and we end up with 256K image (which is admittedly already big for zephyr), this might be well worth a comment.

Reply:
Is it possible to use macros in dts files so that partition size definitions can be changed based on a specific application?